test/main.js

```diff
@@ -344,62 +370,77 @@ describe('gulp-bless', function() {
                 })
             ];
 
-            //note that for each assert that reads large file, we will provide custom error message to speed up the test execution 
-            //shall there be any failure.
-            gulp.src('./test/css/long.css')
-                .pipe($.rename({
-                    suffix: '-split'
-                }))
+            var concatName = "long-split.css";
+            var sourceName = "small.css";
+
+            gulp.src('./test/css/long-parent.scss')
                 .pipe($.sourcemaps.init())
-                .pipe(concat("long-split.css")) //concat will produce a source mapping
+                .pipe(sass({outputStyle: 'expanded'}))
+                .pipe(concat(concatName))
                 .pipe(bless())
+                .pipe(cleanCss({processImport: false, advanced: false}))
                 .pipe($.sourcemaps.write('./generated-sourcemaps'))
                 .pipe(assert.length(4))
                 .pipe(assert.nth(0, function(result) {
                     result.path.should.match(/long-split\.css\.map$/);
-                    var expectedValueFile = './test/css/long-split-with-pre-existing-sourcemap.map';
-                    result.contents.toString('utf8').should.equal(
-                        fs.readFileSync(expectedValueFile).toString('utf8').trim(),
-                        "first split source map does not contain expected content as contained in " + expectedValueFile
-                    );
+                    var sourceMapPart1 = JSON.parse(result.contents.toString('utf8'));
+                    sourceMapPart1.version.should.equal(3);
+                    sourceMapPart1.file.should.equal("../" + concatName);
+                    sourceMapPart1.sources.should.deepEqual([sourceName])
+                    //yes, souce content should be from the small.css which included by parente.scss
```

Comment:
Actually no, I meant `small.css` because `long-parent.scss` just contains lots of `@import` of `small.css` so technically stuff in `small.css` should be the source for everything in the blessed file; stuff in `long-parent.scss` is just container of the actual source.